src/query/storage/m3/multi_fetch_result.go

```diff
@@ -83,6 +83,23 @@ func (r *multiResult) Close() error {
 	return nil
 }
 
+func (r *multiResult) FinalResultWithAttrs() (
+	encoding.SeriesIterators, []storage.Attributes, error) {
+	iters, err := r.FinalResult()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	i := 0
+	attrs := make([]storage.Attributes, iters.Len())
+	for _, res := range r.dedupeMap {
+		attrs[i] = res.attrs
```

Comment:
Hm, we don't always have the dedupe map in the case of we only got one set of iterators.

You'd have to set the `attrs[i] = r.seenFirstAttrs` for that case.

Perhaps check if `len(r.dedupeMap) != iters.Len()` and if so just return `r.seenFirstAttrs` for all of the series?

Reply:
Good call